OptimobileShared/Network/NetworkClient.swift

```diff
@@ -73,10 +75,17 @@ class KSHttpClient {
 
     // MARK: HTTP Methods
 
-    @discardableResult func sendRequest(_ method: KSHttpMethod, toPath: String, data: Any?, onSuccess: @escaping KSHttpSuccessBlock, onFailure: @escaping KSHttpFailureBlock) -> URLSessionDataTask {
-        let request = newRequestToPath(toPath, method: method, body: data)
+    func sendRequest(_ method: KSHttpMethod, toPath: String, data: Any?, onSuccess: @escaping KSHttpSuccessBlock, onFailure: @escaping KSHttpFailureBlock) {
+        var request = newRequestToPath(toPath, method: method, body: data)
 
-        return sendRequest(request: request, onSuccess: onSuccess, onFailure: onFailure)
+        do {
+            try authorization.authorizeRequest(&request)
+        } catch {
+            onFailure(nil, error, nil)
```

Comment:
if the http_client returns an error on attempt to authorize the request, we pass that error to `onFailure` block

Reply:
can we go through what happens to events that failed to sync due to missing header. When will we attempt to sync them again?

it seems to work different VS android
- in android if it fails WorkManager will do a retry with default exponential strategy
- in ios sync happens when AnalyticsHelper is initialised and when app goes to bg (tracking immediate bg event). So, if we did 2nd init seems they would instantly sync --> seems ok

In general, for android and ios we should settle on whether we want flush when creds given or OK to rely on later retry

Reply:
I added a new system notification named `optimobileInializationFinished`, where ddl and analytic services are subscribers. They will try to send a cache to server